lib/ansible/parsing/vault/__init__.py

```diff
@@ -498,9 +498,8 @@ def __init__(self, filename=None, encoding=None, loader=None, vault_id=None):
 
     def _run(self, command):
         try:
-            p = subprocess.Popen(command,
-                                 stdout=subprocess.PIPE,
-                                 stderr=subprocess.PIPE)
+            # STDERR not captured to make it easier for users to prompt for input in their scripts
+            p = subprocess.Popen(command, stdout=subprocess.PIPE)
```

Comment:
This is already possible by prompting on the TTY instead of stderr, and is the mechanism used elsewhere and recommended in this case.

Reply:
Ah, sorry I should have been clearer.
The issue is not the capturing of user input, but that the user gets no message or feedback except an empty prompt.
Or I'm doing something really weird,

I have script that looks in the users keyring, but if nothing is found I want to prompt for a password
`   sys.stderr.write("Dummy message")
    print("testing")
    password = input("Enter password: ")
    print(password)`

The excepted result here would be the message printed  but all I get is an empty line where I can type in the password.

With the patch I get the messages properly.

If I'm using pythons getpass module I actually get the message before the prompt, but I want to give a bit more information than that.

Reply:
Also the documentation I found https://docs.ansible.com/ansible/latest/user_guide/vault.html#storing-passwords-in-third-party-tools-with-vault-password-client-scripts said to use std, I would be happy to update it if it's wrong?